webpack/assets/javascripts/react_app/components/notifications/NotificationDrawerToggle.js

```diff
@@ -21,7 +21,7 @@ class NotificationDrawerToggle extends Component {
     switch (actionType) {
       case ACTIONS.RECEIVED_NOTIFICATIONS: {
         this.setState({
-          count: NotificationsStore.getNotifications().length,
+          unreadNotificationsCount: NotificationsStore.hasUnreadNotifications(),
```

Comment:
for the polling use case perhaps it would be best to keep a running count of unread notifications, reset when notifications are received and adjusted when notification is marked as read. The store data for the read notification should then also be updated (my bad)